rolling_bonxai_server_node:
  ros__parameters:
    frame_id: "map"
    base_frame_id: "zed_camera_link"
    topic_in: "/zed/zed_node/point_cloud/cloud_registered"

    occupancy:
      resolution: 0.02
      occupancy_min_z: -100.0
      occupancy_max_z: 100.0
      occupancy_threshold: 0.50
    
    sensor_model:
      max_range: 10.0
      hit: 0.7
      miss: 0.4
      min: 0.12
      max: 0.97

    rolling:
      coordinate_system:
        chunk_size: 5.0 #double
        
      transition_manager:
        hysteresis_ratio: 0.2 #[0.01,1.0]

      loading_policy:
        name: "neighbourhood" # "planar","temporal" ##String
        neighborhood:
          radius: 2 #int
        planar:
          radius: 2 #int
          z_min: 1 #int
          z_max: 3 #int
          use_relative: true #bool
        temporal:
          radius: 2 #int
          age_weightage: 0.3 #[0.01,1.0] double
          max_age_s: 300 #int in seconds

      storage:
        full_save_folder_path: "/home/tmp/rolling_bonxai"

      asyncio:
        num_load_threads: 3 #int
        num_save_threads: 1 #int

    server:
      cleanup_interval_sec: 300.0

    stats:
      enable_stats: true
      quick_stats: true
      publish_occupied_voxels: true
      publish_rate: 3.0 #Hz